src/sink/send.rs

```diff
@@ -0,0 +1,50 @@
+use {Poll, Async, Future};
+use sink::{Sink, AsyncSink};
+
+/// Future for the `Sink::send` combinator, which sends a value to a sink and
+/// then waits until the sink has fully flushed.
+#[must_use = "futures do nothing unless polled"]
+pub struct Send<S: Sink> {
+    sink: Option<S>,
+    item: Option<S::SinkItem>,
+}
+
+pub fn new<S: Sink>(sink: S, item: S::SinkItem) -> Send<S> {
+    Send {
+        sink: Some(sink),
+        item: Some(item),
+    }
+}
+
+impl<S: Sink> Send<S> {
+    fn sink_mut(&mut self) -> &mut S {
+        self.sink.as_mut().take().expect("Attempted to poll Send after completion")
+    }
+
+    fn take_sink(&mut self) -> S {
+        self.sink.take().expect("Attempted to poll Send after completion")
+    }
+}
+
+impl<S: Sink> Future for Send<S> {
+    type Item = S;
+    type Error = S::SinkError;
+
+    fn poll(&mut self) -> Poll<S, S::SinkError> {
+        if let Some(item) = self.item.take() {
+            try!(self.sink_mut().poll_complete());
```

Comment:
Like with the buffering above, perhaps this could get delayed until after after the start send returned not ready?